src/main/java/io/vertx/core/http/impl/Http1xServerConnection.java

```diff
@@ -76,127 +69,177 @@
 
   private static final Logger log = LoggerFactory.getLogger(Http1xServerConnection.class);
 
-  private static final int CHANNEL_PAUSE_QUEUE_SIZE = 5;
-
   private final String serverOrigin;
   private final SSLHelper sslHelper;
-  final HttpServerOptions options;
   private WebSocketServerHandshaker handshaker;
-  private final HttpServerMetrics metrics;
   private boolean requestFailed;
-  private Object requestMetric;
-  private Handler<HttpServerRequest> requestHandler;
   private Handler<ServerWebSocket> wsHandler;
-  private HttpServerRequestImpl currentRequest;
-  private HttpServerResponseImpl pendingResponse;
   private ServerWebSocketImpl ws;
   private long bytesRead;
   private long bytesWritten;
 
-  // (queueing == true) <=> (paused || pending.size() > 0)
-  private final Deque<Object> pending = new ArrayDeque<>(8);
-  private boolean paused;
-  private boolean sentCheck;
-  private boolean queueing;
+  private HttpServerRequestImpl pendingRequest;  // tail
+  private HttpServerRequestImpl pendingResponse; // head
```

Comment:
it would be incorrect because a request can still be in process and a response be sent already, think an handler that receives a long body but sent a response before having received fully the request. Like above, this is actually the response being sent, so like above I would suggest something like _responseInProgress_

Reply:
I will add more general doc about the behavior of state machine